src/App.tsx

```diff
@@ -1,14 +1,40 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSelectedTodo, setIsSelectedTodo] = useState<Todo | null>(null);
+  const [filterStatus, setFilterStatus] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const getFilteredTodos = todos.filter(todo => {
+    const matchesStatus =
+      filterStatus === 'all' ||
+      (filterStatus === 'active' && !todo.completed) ||
+      (filterStatus === 'completed' && todo.completed);
+
+    const matchesQuery = todo.title.toLowerCase().includes(query.toLowerCase());
+
+    return matchesStatus && matchesQuery;
+  });
```

Comment:
The `getFilteredTodos` should be a function, not a variable. You should define it as a function that returns the filtered todos, like `const getFilteredTodos = () => todos.filter(todo => {...});`. Then, use `getFilteredTodos()` when passing it to the `TodoList` component.